File: AirlineGt/AirlineGt/ClientApp/app/shared/factura.ts
import * as _ from "lodash";
import { IVuelo } from './ivuelo';

export class Factura {
    id: number;
    numeroFactura: string;
    fechaEmision: string;
    pasajero: Pasajero = new Pasajero();
    boletos: Array<Boleto> = new Array<Boleto>(); 
    maleta: number = 0;
    subTotal: number = this.subtotal;
    get subtotal(): number {
        return _.sum(_.map(this.boletos, i => ((i.vuelo.precioTurista * i.cantidad) + this.maleta) ));
    }
} 

export class Boleto {
    id: number;
    vuelo: IVuelo;
    cantidad: number;
   
}

export class Pasajero {
    id: number;
    nombres: string = "";
    apellidos: string;
    fechaNacimiento: string;
    correoElectronico: string;
    numeroTelefonico: number;
    tipoDocumento: string;
    numeroIdentificacion: number;
    domicilio: string;
    direccion: string;
    departamento: string;
    ciudad: string;
    codigoPostal: string;
    tarjeta: Tarjeta = new Tarjeta();
}
export class Tarjeta {
    id: number;
    franquicia: string;
    lugarEmision: string;
    numeroTarjeta: number;
    fechaVencimiento: string;
    ccv: number;
}

File: AirlineGt/AirlineGt/Data/Entities/Factura.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace AirlineGt.Data.Entities
{
    public class Factura
    {
        public int Id { get; set; }
        public string NumeroFactura { get; set; }
        public DateTime fechaEmision { get; set; }
        public Pasajero Pasajero { get; set; }
        public ICollection<Boleto> Boletos{ get; set; }
          = new List<Boleto>();
        public int Maleta { get; set; }
        public double subTotal { get; set; }
    }
}
